--- a/Source/cmake/OptionsGTK.cmake 
+++ b/Source/cmake/OptionsGTK.cmake 
@@ -82,7 +82,9 @@ if (DEVELOPER_MODE)
 else ()
     WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_MINIBROWSER PUBLIC OFF)
     WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_API_TESTS PRIVATE OFF)
-    set(WebKit2_VERSION_SCRIPT "-Wl,--version-script,${CMAKE_MODULE_PATH}/gtksymbols.filter")
+    if (NOT CMAKE_SYSTEM_NAME MATCHES "Darwin")
+        set(WebKit2_VERSION_SCRIPT "-Wl,--version-script,${CMAKE_MODULE_PATH}/gtksymbols.filter")
+    endif ()
 endif ()
 
 if (CMAKE_SYSTEM_NAME MATCHES "Linux")
